src/vaults/vaults.service.ts

```diff
@@ -550,6 +552,63 @@ export class VaultsService extends Service {
     return Strategy__factory.connect(strategyAddress, this.sdk.provider);
   }
 
+  async deposit(vault: string, amount: BigNumber): Promise<TransactionStatus> {
+    if (!this.sdk.address || !this.sdk.signer) {
+      console.error(`Failed deposit to ${vault}, requires an active signer`);
+      return TransactionStatus.Failure;
+    }
+    const vaultContract = Vault__factory.connect(vault, this.sdk.signer);
+    const depositToken = await vaultContract.token();
+    const depositTokenContract = Erc20__factory.connect(
+      depositToken,
+      this.sdk.provider,
+    );
+    const allowance = await depositTokenContract.allowance(
+      this.sdk.address,
+      vault,
+    );
+    if (amount.gt(allowance)) {
+      try {
+        const approveTx = await depositTokenContract.increaseAllowance(
+          vault,
+          amount,
+        );
+        await approveTx.wait();
+      } catch (err) {
+        console.log(err);
+        return TransactionStatus.Failure;
+      }
+    }
+    try {
+      // TODO: try a look up via badger api if possible for the current address proof
+      const depositTx = await vaultContract['deposit(uint256,bytes32[])'](
+        amount,
+        [],
+      );
+      await depositTx.wait();
+      return TransactionStatus.Success;
+    } catch (err) {
+      console.error(err);
+      return TransactionStatus.Failure;
+    }
+  }
+
+  async withdraw(vault: string, amount: BigNumber): Promise<TransactionStatus> {
+    if (!this.sdk.address || !this.sdk.signer) {
```

Comment:
address goes from `this.signer.getAddress();`, do we need to double check?

Reply:
```
  // from sdk init
  private async initialize() {
    try {
      if (this.signer) {
        this.address = await this.signer.getAddress();
      }
    } catch {} // ignore errors from getting address
  }
  
  // from service class intermediary 
  get address(): string | undefined {
    return this.sdk.address;
  }
```
yeah the ordering probably could be changed but since this is set up in this way i think either order is fine and should probably short cricuit on signer missing as address missing w.o signer is an invariant